### PR TITLE
Add documentation examples for `invalid-sequence-index`

### PR DESCRIPTION
**Pull request by julthep**
Co-authored-by: Vladyslav Krylasov <vladyslav.krylasov@gmail.com>

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description
Add documentation examples for `invalid-sequence-index`

Refs #5953

### CODE REVIEW

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]

File: doc/data/messages/i/invalid-sequence-index/bad.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+fruit = ['apple', 'banana', 'orange']
+print(fruit[object])  # [invalid-sequence-index]
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/i/invalid-sequence-index/bad.py:2]:
```suggestion
fruits = ['apple', 'banana', 'orange']
print(fruits["apple"])  # [invalid-sequence-index]
```
We should use an example without a catch imo.

Comment (julthep) [reply doc/data/messages/i/invalid-sequence-index/bad.py]: Yes, agree with you.

File: doc/data/messages/i/invalid-sequence-index/bad.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+fruits = ['apple', 'banana', 'orange']
+print(fruits['apple'])  # [invalid-sequence-index]
+# but be careful with data[True], it will work since True is represnted as 1
+# and False as 0, so it will work
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/i/invalid-sequence-index/bad.py:4]: You can create a details.rst to add this information is you want. If you think pylint should raise in this case (arguably yes, then it's a false negative that we could fix).

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]

**Review by Pierre-Sassoulas — APPROVED** [review]

File: doc/data/messages/i/invalid-sequence-index/bad.py (hunk 2 above)
Comment (julthep) [reply doc/data/messages/i/invalid-sequence-index/bad.py]: Okey, so let's move them to `details.rst`. Thank you very much.

**Pull request merged**